### PR TITLE
add PROJECT_CHARTER

### PR DESCRIPTION
**Pull request by Eomm**
- [ ] Project Charter is published on website or github

I followed [this template](https://github.com/openjs-foundation/cross-project-council/blob/master/PROJECT_CHARTER_TEMPLATE.md)

I think we should write down a process on the `Section 4.2: Decision-making, Voting, and/or Elections` since there is not a written standard right now in our CONTRIBUTING.md.

Ref https://github.com/openjs-foundation/project-onboarding/issues/35

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by delvedor** [comment]
Please rebase this pr against the new `master` branch, see https://github.com/fastify/fastify/issues/2056#issuecomment-612916945.

File: PROJECT_CHARTER.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+The Fastify project is part of the [OpenJS Foundation], which operates transparently, openly, collaboratively, and ethically. Project proposals, timelines, and status must not merely be open, but also easily visible to outsiders.
+
+
+## Section 1: Scope
+
+Fastify is a web framework highly focused on providing the best developer experience with the least overhead and a plugin architecture.
+
+### 1.1: In-scope
+
++ Code updates and improvementes focused on performance
```
Comment (mcollina) [inline PROJECT_CHARTER.md:17]:
```suggestion
+ Develop a web framework for Node.js with a focus on developer experience, performance and extensibility.
```

File: PROJECT_CHARTER.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
++ Plugin Architecture
++ Support web protocols
++ Tested codebase
```
Comment (mcollina) [inline PROJECT_CHARTER.md:20]:
```suggestion
+ Official plugins for common user requirements
```

**Comment by Eomm** [comment]
> I think we should write down a process on the Section 4.2: Decision-making, Voting, and/or Elections since there is not a written standard right now in our CONTRIBUTING.md.

@mcollina Is it ok let this section blank?

**Comment by mcollina** [comment]
> **I think we should write down a process on the Section 4.2: Decision-making, Voting, and/or Elections since there is not a written standard right now in our CONTRIBUTING.md.**

I think it's ok to leave this blank. Is it correct @jorydotcom?

File: PROJECT_CHARTER.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
+# Fastify Charter
+
+The Fastify project aims to build a fast and low overhead web framework, for Node.js.
```
Comment (jsumners) [inline PROJECT_CHARTER.md:3]:
```suggestion
The Fastify project aims to build a fast and low overhead web framework for Node.js.
```

File: PROJECT_CHARTER.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+
+## Section 0: Guiding Principles
+
+The Fastify project is part of the [OpenJS Foundation], which operates transparently, openly, collaboratively, and ethically. Project proposals, timelines, and status must not merely be open, but also easily visible to outsiders.
```
Comment (jsumners) [inline PROJECT_CHARTER.md:8]: Link is incomplete.

File: PROJECT_CHARTER.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+  + Framework API
+  + Website
++ Write easier APIs for developers
++ Tools:
+  + CI services
+  + Bots to improve overall efficiency
+  + Releases
++ Support:
+  + Community
+  + User's issue and questions
```
Comment (jsumners) [inline PROJECT_CHARTER.md:33]:
```suggestion
  + Users's issues and questions
```

File: PROJECT_CHARTER.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
++ Contributor's pull request review
```
Comment (jsumners) [inline PROJECT_CHARTER.md:34]:
```suggestion
+ Contributors's pull request review
```

File: PROJECT_CHARTER.md [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+
+### 1.2: Out-of-Scope
+
++ Support versions of Node.js at EOF (end of life) stage
```
Comment (jsumners) [inline PROJECT_CHARTER.md:38]:
```suggestion
+ Support versions of Node.js at EOL (end of life) stage
```

File: PROJECT_CHARTER.md [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
++ Support serverless architecture
++ Contributions that violates the [Code of Conduct](./CODE_OF_CONDUCT.md)
+
+
+## Section 2: Relationship with OpenJS Foundation CPC.
+
+Technical leadership for the projects within the [OpenJS Foundation] is delegated to the projects through their project charters by the [OpenJS Foundation Cross-Project Council](https://openjsf.org/about/governance/) (CPC). In the case of the Fastify project, it is delegated to the [Fastify's Team](./README.md#team) (FTM). The OpenJS Foundation's business leadership is the Board of Directors (the "Board").
```
Comment (jsumners) [inline PROJECT_CHARTER.md:45]: Broken link.

**Comment by jorydotcom** [comment]
@Eomm @mcollina for that section you can write "Section Intentionally Left Blank" or alternatively, you can say simply that decisions are made cooperatively between the lead maintainers.

**Comment by mcollina** [comment]
@Eomm this should have had approval by the OpenJS Foundation CPC before landing :)

**Comment by Eomm** [comment]
Oh sorry, I didn't understand 🙏
I can't revert it on master

Would be OK to open a pr to delete it and a new one to add it?

**Comment by mcollina** [comment]
Let's leave it like this and see if the CPC needs some changes first.

File: PROJECT_CHARTER.md [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
++ Contributions that violates the [Code of Conduct](CODE_OF_CONDUCT.md)
+
+
+## Section 2: Relationship with OpenJS Foundation CPC.
+
+Technical leadership for the projects within the [OpenJS Foundation][openjs foundation] is delegated to the projects through their project charters by the [OpenJS Foundation Cross-Project Council](https://openjsf.org/about/governance/) (CPC). In the case of the Fastify project, it is delegated to the [Fastify's Team](README.md#team) (FTM). The OpenJS Foundation's business leadership is the Board of Directors (the "Board").
+
+This Fastify Charter reflects a carefully constructed balanced role for the FTM and the CPC in the governance of the OpenJS Foundation. The charter amendment process is for the FTM to propose changes using simple majority of the full FTM, the proposed changes being subject to review and approval by the CPC. The CPC may additionally make amendments to the FTM charter at any time, though the CPC will not interfere with day-to-day discussions, votes or meetings of the FTM.
+
+ex. [Node.js TSC Charter](https://github.com/nodejs/TSC/blob/master/TSC-Charter.md#section-2-evolution-of-openjs-foundation-governance)
```
Comment (eemeli) [inline PROJECT_CHARTER.md:49]: I think this line's been left here accidentally from the charter template?

Comment (mcollina) [reply PROJECT_CHARTER.md:49]: Good spot, we'll fix!

**Pull request merged**